src/methods/proxy/announce.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Args,
+  BaseTxInfo,
+  createMethod,
+  OptionsWithMeta,
+  UnsignedTransaction,
+} from '../../util';
+
+interface ProxyAnnounceArgs extends Args {
+  /**
+   * The account that the proxy will make a call on behalf of.
+   */
+  real: string;
+  /**
+   * The hash of the call to be made by the `real` account.
+   */
+  callHash: string;
+}
+
+/**
+ * Publish the hash of a proxy-call that will be made in the future.
+ *
+ * This must be called some number of blocks before the corresponding `proxy` is attempted
```

Comment:
"some number of b locks": is there a minimum required?

Reply:
I think this refers to the delay period: https://github.com/paritytech/substrate/blob/master/frame/proxy/src/lib.rs#L137 (but I am not certain, the inline docs of the proxy pallet seem to have only been partially updated with the last overhaul of the pallet). I will try and update to explain better.